### PR TITLE
Fixes #23039: Add rich form types to yaml techniques

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/23039

For the most part, adds support for all directive form niceties supported in the built-in techniques.

Screenshot from 2023-07-19 15-06-48

A few interesting points:
* It does not stick to the model used in the `metadata.xml`. Particularly we use types as much as possible, and less constraints. Some values were renamed for either consistency with what is already there or just clarity.
* The escaping behavior is not part of the type but an independent parameter, allowing more flexibility (and hopefully less confusion).
* Select and regex parameter types have been merged with the ones used for method parameters, in order to converge iteratively.
* Removes `ParameterType` (scala)/`Escaping` (rust) from technique parameters, this was a mistake we only have them on method parameters for now.
* Allow omitting ids: there are generating on the fly during compilation. Also adding a command allowing to add ids to an existing files (though for now it removes all formatting/comments).

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/rudderc/tests/cases/general/escaping/metadata.xml [hunk 1]
```diff
@@ -33,12 +33,10 @@
...
       <INPUT>
-        <NAME>3439BBB0-D8F1-4C43-95A9-0C56BFB8C27E</NAME>
-        <DESCRIPTION>server</DESCRIPTION>
-        <LONGDESCRIPTION>${sys.host} . | / ${sys.${host}} &apos; &apos;&apos; &apos;&apos;&apos; $ $$ &quot; &quot;&quot; \ \\😋aà3</LONGDESCRIPTION>
+        <NAME>server</NAME>
+        <DESCRIPTION>${sys.host} . | / ${sys.${host}} &apos; &apos;&apos; &apos;&apos;&apos; $ $$ &quot; &quot;&quot; \ \\😋aà3</DESCRIPTION>
         <CONSTRAINT>
-          <TYPE>textarea</TYPE>
-          <MAYBEEMPTY>true</MAYBEEMPTY>
+          <TYPE>string</TYPE>
```
Comment (Fdall) [inline policies/rudderc/tests/cases/general/escaping/metadata.xml:39]: Why the change from textarea to string here?

Comment (amousset) [reply policies/rudderc/tests/cases/general/escaping/metadata.xml:39]: because previous code used to put textarea everywhere

**Comment by amousset** [comment]
PR updated with a new commit
